sdk/src/main/scala/com/horizen/SidechainApp.scala

```diff
@@ -107,6 +107,7 @@ class SidechainApp @Inject()
       sidechainId = BytesUtils.fromHexString(sidechainSettings.genesisData.scId),
       sidechainGenesisBlockId = genesisBlock.id,
       genesisMainchainBlockHash = genesisBlock.mainchainHeaders.head.hash,
+      preGenesisMainchainBlockHash = genesisBlock.mainchainHeaders.head.hash,
```

Comment:
I prefer to leave it as is, the proposed name `genesisMainchainBlockParentHash` is good but it could be hard to distinguish `genesisMainchainBlockHash` and `genesisMainchainBlockParentHash`, but it easy to distinguish  `genesisMainchainBlockHash` and `preGenesisMainchainBlockHash`

Reply:
Honestly, I don't like "pre" at all. Because it means "previous genesis mc block hash", that is not about block before - parent block, but about another genesis block, which is misleading here.

Reply:
`genesisBlock.mainchainHeaders.head.hash` an error here -> must be `...head.hashPrevBlock`